### DISCUSSION TITLE
I want to understand how MetaMask (mobile app) and our Dapp establish sessions.

### DISCUSSION THREAD

**Discussion by vaxberry-ash** [WalletConnect/walletconnect-monorepo#5570] [Developer Support]
I recently started using ReownSDK for the first time. In a desktop browser environment, I don’t encounter any issues while sending transactions to the blockchain network. The problem lies in the mobile environment. When communicating between MetaMask and the Dapp I developed, it’s extremely difficult to pinpoint issues if something goes wrong. To make accurate assumptions about the problem, I believe I need a clear understanding of how the session is established. Could you explain how sessions are established?

**Comment by arein** [comment]
Please visit our docs and specs. MetaMask are shortly refactoring their current single-chain implementation on mobile to support the multichain logic all other wallets support. We expect the reliability of MetaMask Mobile to then match other wallets.

**Reply by Rsed19901** [reply]
> Please visit our docs and specs. MetaMask are shortly refactoring their current single-chain implementation on mobile to support the multichain logic all other wallets support. We expect the reliability of MetaMask Mobile to then match other wallets.

**Reply by Rsed19901** [reply]
> Please visit our docs and specs. MetaMask are shortly refactoring their current single-chain implementation on mobile to support the multichain logic all other wallets support. We expect the reliability of MetaMask Mobile to then match other wallets.
